amplify/backend/api/colonycdapp/schema.graphql

```diff
@@ -297,14 +297,21 @@ type User @model {
 }
 
 type Domain @model {
-  id: ID! @index(name: "byNativeId", sortKeyFields: ["nativeId"]) # colonyAddress_nativeId
+  id: ID! # We have to self-manage this and keep it the following format: colonyAddress_nativeId
+  colonyId: ID! @index(name: "byColony")
+  colony: Colony! @belongsTo(fields: ["colonyId"])
   nativeId: Int!
   nativeFundingPotId: Int!
   nativeSkillId: Int!
-  name: String
-  description: String
-  color: DomainColor
-  parent: Domain @hasOne
+  isRoot: Boolean!
+  metadata: DomainMetadata @hasOne(fields: ["id"])
+}
+
+type DomainMetadata @model {
+  id: ID! # This field is referenced by Domain id, so has to be in the same format: colonyAddress_nativeId
+  name: String!
+  description: String!
+  color: DomainColor!
```

Comment:
Separated this since it's a piece of information that's only available in the CDapp (and doesn't get emitted to the network, therefore inaccessible in the block-ingestor). This way, the saga is only responsible for creating the metadata, and the actual domain entity is created by action handler in block-ingestor

The advantage this gives is that block-ingestor will still pick up the events that originated from a contract call as opposed to a saga inside CDapp. Those will not have any metadata attached, but will still show correctly in the actions list.

We can follow the same pattern in the colony creation saga